R/convert_data.R

```diff
@@ -48,6 +48,13 @@
     )
   }
 
+  if (is_sparse_tibble(data)) {
+    cli::cli_abort(
+      "Sparse data cannot be used with formula interface. Please use 
+      {.fn fit_xy} instead."
+    )
```

Comment:
```suggestion
    cli::cli_abort(c(
      "Sparse data cannot be used with formula interface by default.",
      "i" = "Use {.fn fit_xy} instead to preserve sparsity or set the option
             {.help [{.code sparsevctrs.verbose_materialize = NULL}](TODO::LINK_HERE)}
             to allow for materializing sparse vectors."
    ))
```

I think we ought to let people know they can bypass this restriction in the error message. Is there a help-page documenting this option that we can link to?

Reply:
I'm setting `sparsevctrs.verbose_materialize` in the tests to make sure things work correctly. But the option is generally not meant to be used by the user. Sparse vectors will materialize silently when needs be. So you don't have to set any options.

I can be less opinionated and switch to a warning here instead 😄

Reply:
Okay, two thoughts on revisiting:

1) I agree that the default behavior should be to error here. No need to convert to warning. :)
2) Before, I thought it'd be nice if users could get around this error and allow for materializing sparse data in `fit()`, but now that I'm looking at this again I'm realizing that users won't have the analogous option for workflows, so I'm fine with ignoring this comment and keeping that option developer-facing only.